schema.json

```diff
@@ -0,0 +1,488 @@
+{
+    "$schema": "https://json-schema.org/draft-07/schema",
+    "type": "object",
+    "properties": {
+        "copyright": {
+            "type": "string",
+            "description": "The license string to include at the top of the generated header"
+        },
+        "enum_prefix": {
+            "type": "string",
+            "description": "The dedicated enum prefix for the implementation specific header to avoid collisions"
+        },
+        "constants": {
+            "type": "array",
+            "items": {
+                "type": "object",
+                "properties": {
+                    "name": {
+                        "type": "string",
+                        "description": "Name of the constant variable/define"
+                    },
+                    "value": {
+                        "type": "string",
+                        "description": "An unsigned integer in decimal or hexadecimal syntax, in string"
```

Comment:
Regex patterns are useful here too: `^(0x[0-9A-F]{4}|0|[1-9][0-9]{0,4})$` or something like that

Though perhaps better - since we're using YAML already which supports hexadecimal numbers - could we just accept integers here instead? With range validation:
https://json-schema.org/understanding-json-schema/reference/numeric#range
(Can also be future work)